src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,131 @@
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd(movie: Movie): void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+  const [isImdbUrlWrong, setIsImdbUrlWrong] = useState(false);
+  const [isImgUrlWrong, setIsImgUrlWrong] = useState(false);
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
```

Comment:
DRY (Do not Repeat Your self) - create const out of component with empty state and use it where needs
```
const EMPTY_MOVIE = {
  title: '',
  description: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: '',
}
````

```suggestion
    setNewMovie(EMPTY_MOVIE);
```